kernel/src/engine/mod.rs

```diff
@@ -5,26 +5,32 @@
 #[cfg(feature = "arrow-conversion")]
 pub(crate) mod arrow_conversion;
 
-#[cfg(feature = "arrow-expression")]
+#[cfg(all(
+    feature = "arrow-expression",
+    any(feature = "default-engine", feature = "sync-engine")
+))]
 pub mod arrow_expression;
 
-#[cfg(any(feature = "default-engine", feature = "sync-engine"))]
```

Comment:
Moved this to a macro to avoid the repeated checking of the two feature flags. Hopefully reviewers agree it's better :)